test/Bridge/Symfony/Validator/Constraint/EnumTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Greg0ire\Enum\Tests\Bridge\Symfony\Validator\Constraint;
+
+use Greg0ire\Enum\Bridge\Symfony\Validator\Constraint\Enum;
+
+final class EnumTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider getValidEnum
+     */
+    public function testValidEnumClasses($enumClass)
+    {
+        $enumConstraint = new Enum($enumClass);
+
+        $this->assertSame($enumClass, $enumConstraint->class);
+    }
+
+    public function getValidEnum()
+    {
+        return array(
+            array('Greg0ire\Enum\Tests\Fixtures\AllEnum'),
+            array('Greg0ire\Enum\Tests\Fixtures\DummyEnum'),
+            array('Greg0ire\Enum\Tests\Fixtures\FooEnum'),
+        );
+    }
+
+    /**
+     * @dataProvider getInvalidEnum
+     * @expectedException \Symfony\Component\Validator\Exception\ConstraintDefinitionException
+     */
+    public function testInvalidEnumClasses($enumClass)
+    {
+        new Enum($enumClass);
+    }
+
+    public function getInvalidEnum()
+    {
+        return array(
+            array('Greg0ire\Enum\Tests\Fixtures\FooInterface'),
```

Comment:
Not really, but it's prepared for that. :-)

Do you want removing of the provider?